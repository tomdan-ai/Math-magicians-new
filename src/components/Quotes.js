import React, { useState, useEffect } from 'react';
import axios from 'axios';

function Quotes() {
  const [quote, setQuote] = useState(null);
  const [author, setAuthor] = useState('');
  const [isLoading, setIsLoading] = useState(true);
  const [error, setError] = useState(null);

  useEffect(() => {
    axios.get('https://api.api-ninjas.com/v1/quotes?category=intelligence', {
      headers: {
        'X-Api-Key': 'VbJyG2AVoyP/jcAXQ2lvMA==rXycyQZaoKGMz0AG',
      },
    })
      .then((response) => {
        setQuote(response.data[0].quote);
        setAuthor(response.data[0].author);
        setIsLoading(false);
      })
      .catch((error) => {
        setError(error);
        setIsLoading(false);
      });
  }, []);

  if (isLoading) {
    return (
      <div className="loaders">
        Loading
        <i className="fa fa-spinner" aria-hidden="true" />
      </div>
    );
  }

  if (error) {
    return (
      <div>
        Error:
        {error.message}
      </div>
    );
  }

  return (
    <div>
      <h2 className="Quote-head">Quote:</h2>
      <p className="Quote-body">{quote}</p>
      <p className="Quote-author">
        -
        {author}
      </p>
    </div>
  );
}

export default Quotes;
