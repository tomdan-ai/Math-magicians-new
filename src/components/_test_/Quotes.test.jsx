import { render, screen, waitFor } from '@testing-library/react';
import renderer from 'react-test-renderer';
import axios from 'axios';
import { BrowserRouter } from 'react-router-dom';
import Quotes from '../Quotes';

jest.mock('axios', () => ({
  get: jest.fn(),
}));

describe('Quotes', () => {
  it('shows quote and author', async () => {
    const dummyData = [
      {
        quote: 'The will of man is his happiness.',
        author: 'Friedrich Schiller',
        category: 'happiness',
      },
    ];

    axios.get.mockResolvedValue({ data: dummyData });

    render(
      <BrowserRouter>
        <Quotes />
      </BrowserRouter>,
    );

    await waitFor(() => {
      expect(screen.getByTestId('quote')).toHaveTextContent('The will of man is his happiness.');
      expect(screen.getByTestId('author')).toHaveTextContent('Friedrich Schiller');
    });
  });
});

describe('Quotes snapshot', () => {
  it('shows Quotes snapshot', () => {
    const tree = renderer.create(
      <BrowserRouter>
        <Quotes />
      </BrowserRouter>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
