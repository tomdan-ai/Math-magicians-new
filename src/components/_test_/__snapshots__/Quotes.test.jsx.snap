// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quotes snapshot shows Quotes snapshot 1`] = `
<div
  className="loaders"
  data-itemid="load"
>
  Loading
  <i
    aria-hidden="true"
    className="fa fa-spinner"
  />
</div>
`;
